dm/syncer/data_validator.go

```diff
@@ -489,13 +495,38 @@ func (v *DataValidator) doValidate() {
 		default:
 			currLoc.Position.Pos = e.Header.LogPos
 		}
-
+		// update validator metric
+		metrics.ValidatorBinlogPos.WithLabelValues(v.cfg.Name, v.cfg.SourceID, v.cfg.WorkerName).Set(float64(currLoc.Position.Pos))
```

Comment:
Would you give me some hints about asynchronously updating (e.g. put them into an independent go routine)?

Reply:
just a litte suggention ok to not change
maybe add new job type `metrics` and use a new worker thread to update metric like we handle `dmljob`

> What about making them asynchronous as a whole 

lgtm ptal @lance6716

Reply:
https://github.com/pingcap/tiflow/pull/5176 the problem is not related to asynchronous.

Reply:
>  Metrics caches the metrics variables to avoid looking up by name in metricsproxy each time.

oooo, the hot path is `WithLabelValues` , a cached var is enough and no need do this in async